### PR TITLE
[API] Remove non-API top Level function imports

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?

* These APIs are not part of the [documented public API](https://github.com/ray-project/ray/blob/master/doc/source/package-ref.rst).
* They are still usable, but just through something like `ray.<module>.<function>`

## Related issue number

* Depends on https://github.com/ray-project/xgboost_ray/pull/89

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/__init__.py [hunk 1]
```diff
@@ -98,29 +95,22 @@
     "_config",
     "get_runtime_context",
     "actor",
-    "actors",
     "workers",
```
Comment (AmeerHajAli) [inline python/ray/__init__.py:98]: Should we remove this one as well?

**Review by AmeerHajAli — APPROVED** [review]
LGTM.
1 minor comment.

**Comment by robertnishihara** [comment]
Looks fine to me as well.

**Review by AmeerHajAli — CHANGES_REQUESTED** [review]
In retrospect, can we import the list and assert its contents in a unit test with a documentation in the unit test for why it should not be modified unless it is a public API that is supported with Ray Client as well.

File: python/ray/__init__.py (hunk 1 above)
Comment (AmeerHajAli) [reply python/ray/__init__.py:98]: It seems like "workers" does not even exist in this file lets get rid of it?

Comment (ijrsvt) [inline python/ray/__init__.py:98]:
```suggestion
```

Good call to remove this, it does not exist!

**Comment by pcmoritz** [comment]
+1 on API changes

File: python/ray/tests/test_top_level_api.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        # TODO(ilr) either move this to PYTHON_API or remove.
+        "get_actor",
+        # In the RuntimeContext API
+        "get_runtime_context",
+    ]
+
+    functions = getmembers(ray, isfunction)
+    function_names = [f[0] for f in functions]
+    assert set(function_names) == set(PYTHON_API + OTHER_ALLOWED_FUNCTIONS)
+
```
Comment (ijrsvt) [inline python/ray/tests/test_top_level_api.py:40]: @AmeerHajAli Here is the test!

Comment (AmeerHajAli) [reply python/ray/tests/test_top_level_api.py:40]: This is really good. Lets add a comment for why we are testing this and making it clear that adding 1 more public API requires supporting it in Ray Client.

**Review by AmeerHajAli — APPROVED** [review]

File: python/ray/tests/test_top_level_api.py (hunk 2 above)
Comment (ijrsvt) [reply python/ray/tests/test_top_level_api.py:40]: Sure, I'll mention that here. This did make me wonder, should we add client support for the Cross Language API?

**Comment by ijrsvt** [comment]
@krfricke @richardliaw Since this required a change to `xgboost_ray`, how do I update the pinned version?

**Comment by krfricke** [comment]
@ijrsvt, is it acceptable if I make a new xgboost_ray release tomorrow and update the pinned version after that?

**Comment by robertnishihara** [comment]
@ijrsvt I created this draft PR to show what I think the top level API should look like (I don't think it should be done in this PR, but posting because they are related in spirit). https://github.com/ray-project/ray/pull/15503

**Comment by krfricke** [comment]
We released XGBoost-Ray 0.0.5 and I created a PR to update the dependency: https://github.com/ray-project/ray/pull/15507

**Comment by ijrsvt** [comment]
@robertnishihara that looks great, this is definitely a good goal to move towards.

**Review by pcmoritz — APPROVED** [review]

**Comment by ijrsvt** [comment]
Mac Wheels are known to have problems, so they are not related to this PR. Furthermore, this failure is related to `pip`:

```
Traceback (most recent call last):
  File "<stdin>", line 24298, in <module>
  File "<stdin>", line 139, in main
  File "<stdin>", line 115, in bootstrap
  File "<stdin>", line 96, in monkeypatch_for_cert
  File "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpsvjb808t/pip.zip/pip/_internal/commands/__init__.py", line 9, in <module>
  File "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpsvjb808t/pip.zip/pip/_internal/cli/base_command.py", line 12, in <module>
  File "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpsvjb808t/pip.zip/pip/_internal/cli/cmdoptions.py", line 30, in <module>
  File "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/tmpsvjb808t/pip.zip/pip/_internal/utils/hashes.py", line 2, in <module>
ImportError: cannot import name 'NoReturn'

```

**Pull request merged**